### ISSUE TITLE
Doesn't adjust scale properly when moving from a HiDPI monitor to a normal monitor (Win10)

### ISSUE THREAD

**Issue by spinda** [keepassxreboot/keepassxc#221]
## Expected Behavior

When I move KeePassXC from my HiDPI monitor to a normal (1x scale) monitor, the main window should adjust scale accordingly. KeePassX handles this fine.

## Current Behavior

KeePassXC ends up huge when moved to the normal monitor.

![image](https://cloud.githubusercontent.com/assets/7958605/22321390/2ef7655a-e349-11e6-8694-7773ca3cab9b.png)

On the left is KeePassX and on the right is KeePassXC. Both windows are the same size when on my HiDPI display, but when I move them to the normal display KeePassX adjusts its scale properly while KeePassXC does not.

## Steps to Reproduce (for bugs)

1. Start KeePassXC with both a HiDPI monitor and a normal (1x) monitor plugged in.
2. If KeePassXC initially opens on the HiDPI display, move it to the normal display.
3. Either way, KeePassXC will end up 2x the size it should be on the 1x display.

## Context

I always put KeePassX on my normal monitor. This issue prevents me from switching to KeePassXC.

## Your Environment

* KeePassXC version/commit used: 2.1.0

  ![image](https://cloud.githubusercontent.com/assets/7958605/22321417/7f7ee8ae-e349-11e6-9ea6-ca9c4c88a319.png)

* Qt version (e.g. Qt 5.3): 5.6.2
* Compiler (e.g. Clang++3.6.0): Whatever you used to compile the Windows 64-bit build on your website
* Operating System and version: Windows 10

**Comment by phoerious** [comment]
I submitted a patch to the hotfix/221-high-dpi-scaling branch which enables High DPI scaling. Would you be able to test it for me?

**Comment by spinda** [comment]
Unfortunately this doesn't seem to have fixed it. I'm still experiencing the same behavior with a patched build (compiled under the MinGW64 environment, checked out from hotfix/221-high-dpi-scaling).

**Comment by phoerious** [comment]
Which Qt version did you use to compile? It must be Qt >= 5.6, otherwise it won't work.
You can also try running the KeePassXC.exe binary from a terminal and set the QT_SCALE_FACTOR environment variable manually as described in https://blog.qt.io/blog/2016/01/26/high-dpi-support-in-qt-5-6/ and see if that works.

**Comment by spinda** [comment]
According to MSYS:

```
$ pacman -Qs qt5
local/mingw-w64-x86_64-qt5 5.6.2-4 (mingw-w64-x86_64-qt mingw-w64-x86_64-qt5)
    A cross-platform application and UI framework (mingw-w64)
```

I've tried running the binary from the website with `QT_SCALE_FACTOR=2`, but that just doubles the size on both of my monitors, so on my normal monitor KeePassXC becomes 4x. `QT_SCALE_FACTOR=1` results in the original reported behavior.

**Comment by phoerious** [comment]
You can specify individual scaling factors separated by semicolons. For instance:
QT_SCALE_FACTOR=1;2

**Comment by spinda** [comment]
I tried setting `QT_AUTO_SCREEN_SCALE_FACTOR=1`, and that makes things better, but not perfect.

![image](https://cloud.githubusercontent.com/assets/7958605/22409123/e8274b7e-e639-11e6-963b-7ff57de66bd0.png)

(with `QT_AUTO_SCREEN_SCALE_FACTOR=1` on the left, without on the right)

The main GUI elements end up the correct size on the normal monitor, but the title bar is still double the correct size. Also, the window size seems somehow out of sync when I drag KeePassXC between monitors. If I shrink KeePassXC down on the normal monitor, then drag it over to the HiDPI monitor, then drag it back to the normal monitor, it ends up at twice the size as before I dragged it over to the HiDPI monitor.

Setting `QT_SCALE_FACTOR=2;1` (my HiDPI monitor is my primary display) produces the same effects.

I'm really curious as to what is different in KeePassXC that causes these issues, when they're not present at all in KeePassX.

**Comment by phoerious** [comment]
KeePassX (at least the last officially released version) is built with Qt4 which doesn't really handle DPI scaling at all. Qt5, on the other hand, does starting with 5.4 or 5.6, respectively. So I guess Windows just took over and scaled it for you. This is usually less than ideal, but in this case Qt seems to be smarter than it is.

Could you maybe try replacing
```
QApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
```

in the main.cpp with

```
QApplication::setAttribute(Qt::AA_DisableHighDpiScaling);
```

and build it again?

**Comment by spinda** [comment]
Trying with

```c++
QApplication::setAttribute(Qt::AA_DisableHighDpiScaling);
```

yields the originally-reported behavior.

Afterward, I tried again with

```c++
QApplication::setAttribute(Qt::AA_DisableHighDpiScaling);
```

and this time it produces the same behavior as `QT_AUTO_SCREEN_SCALE_FACTOR=1`. I'm not sure what changed to make that happen.

**Comment by phoerious** [comment]
I have to postpone this issue to the next release. Unfortunately, we won't find a solution anymore in time.

**Comment by spinda** [comment]
Understood. I appreciate your help in trying to get to the bottom of this.

**Comment by phoerious** [comment]
The 2.1.1 release was built with Qt 5.8 instead of 5.6. Could you try if that made things any better? Please also try with `QT_AUTO_SCREEN_SCALE_FACTOR=1` if it didn't.

**Comment by ghaabor** [comment]
@phoerious I had the same issue on Fedora 25 but with passing `QT_AUTO_SCREEN_SCALE_FACTOR=1` it's working properly and scales up the UI.

**Comment by phoerious** [comment]
Then I suppose setting `QApplication::setAttribute(Qt::AA_EnableHighDpiScaling);` is the only thing we can do here. I'll prepare a commit for the 2.1.2 release branch.

**Comment by phoerious** [comment]
I filed a PR, so we have at least a somewhat working solution. The fix will be merged into 2.1.2.
